File: leetcode_75/dp_1d/min_cost_climbing_stairs_746.py
from types import List
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        top = len(cost) +1
        dp = [0]*top

        for i in range(2, top):
            dp[i] = min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2])

        print(dp)


# 這不會算是greed（會向後看），dp是向前看，

File: leetcode_75/dp_1d/min_cost_climbing_stairs_746.java
// package leetcode_75.dp_1d;

class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int top = cost.length + 1;
        int[] dp = new int[top];

        for (int i = 2; i < top; i++) {
            dp[i] = Math.min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2]);
        }
        return dp[top - 1];
    }
}